modules/tensor_mechanics/src/bcs/TwistingMomentBC.C

```diff
@@ -0,0 +1,128 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "TwistingMomentBC.h"
+#include "Function.h"
+
+registerMooseObject("TensorMechanicsApp", TwistingMomentBC);
+registerMooseObject("TensorMechanicsApp", ADTwistingMomentBC);
+
+template <bool is_ad>
+InputParameters
+TwistingMomentBCTempl<is_ad>::validParams()
+{
+  InputParameters params = TwistingMomentBCParent<is_ad>::validParams();
+  params.addClassDescription("Apply a moment around a pivot point. This should operate on "
+                             "the displaced mesh for large deformations.");
+  MooseEnum component("x=0 y=1 z=2");
+  params.addRequiredParam<Point>("origin", "Pivot point for the bending moment");
+  params.addRequiredParam<RealVectorValue>("direction", "Torque vector");
```

Comment:
I initially found the "direction" and "factor" parameters kind of confusing, although it made sense once I thought about it a bit. I think I'd call them "direction" and "torque", and make them both required. I

I'd use these parameter names and descriptions:
"direction": "Direction of the torque vector. The supplied vector is normalized if not a unit vector.""
"torque": "Time-dependent function defining the magnitude of the torque"

Also, normalize the user-supplied torque vector in the constructor.

Reply:
Hm, that would make it a lot more difficult to apply a specified torque vector. The user would have to compute the magnitude and specify that again as the `torque` parameter